.github/workflows/conformance-datapath.yaml

```diff
@@ -0,0 +1,268 @@
+          echo "::set-output name=base::$(jq -r '.base.sha' pr.json)"
+          echo "::set-output name=head::$(jq -r '.head.sha' pr.json)"
+      - name: Check code changes
+        if: ${{ github.event.issue.pull_request }}
+        uses: dorny/paths-filter@b2feaf19c27470162a626bd6fa8438ae5b263721
+        id: tested-tree
+        with:
+          base: ${{ steps.pr.outputs.base }}
+          ref: ${{ steps.pr.outputs.head }}
+          filters: |
+            src:
+              - '!(test|Documentation)/**'
+
+  setup-and-test:
+    runs-on: self-hosted
+    needs: check_changes
+    name: Setup & Test
+    if: |
+      (github.event_name == 'issue_comment' && (
+        github.event.comment.body == '/ci-datapath' ||
+        (github.event.comment.body == '/test' && needs.check_changes.outputs.tested == 'true')
+      )) ||
+      github.event_name == 'schedule' ||
+      github.event_name == 'pull_request'
+    timeout-minutes: 45
+    steps:
+      - name: Set up job variables
+        id: vars
+        run: |
+          if [ ${{ github.event.issue.pull_request || github.event.pull_request }} ]; then
+            PR_API_JSON=$(curl \
+              -H "Accept: application/vnd.github.v3+json" \
+              -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" \
+              ${{ github.event.issue.pull_request.url || github.event.pull_request.url }})
+            SHA=$(echo "$PR_API_JSON" | jq -r ".head.sha")
+          else
+            SHA=${{ github.sha }}
+          fi
+
+          echo ::set-output name=sha::${SHA}
+
+      - name: Set commit status to pending
+        uses: Sibz/github-status-action@650dd1a882a76dbbbc4576fb5974b8d22f29847f
+        with:
+          authToken: ${{ secrets.GITHUB_TOKEN }}
+          sha: ${{ steps.vars.outputs.sha }}
+          context: ${{ github.workflow }}
+          description: Cilium Datapath Conformance test in progress...
+          state: pending
+          target_url: ${{ env.check_url }}
+
+      - name: Checkout pull request for Helm chart
+        uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b
+        with:
+          ref: ${{ steps.vars.outputs.sha }}
+          persist-credentials: false
+
+      - name: Fetch cilium-cli
+        id: cilium-cli
+        uses: shrink/actions-docker-extract@v1
+        with:
+          # This is hack. Under the hood, the "action-docker-extract" does
+          # "docker cp $(docker create $image):/$path ...". While the cli
+          # image doesn't have an entrypoint, the create cmd fails. So, in
+          # addition to the image, pass the "ls" which is used as the
+          # entrypoint. The hack will go away, once @brb will create a custom
+          # GH action based on https://git.sr.ht/~motiejus/undocker.
+          # Also, we rely on the ":latest" since the datapath conformance suite
+          # is heavily WIP.
+          image: 'quay.io/cilium/cilium-cli-ci:latest ls'
```

Comment:
Why `:latest`? shouldn't we pin a particular version? As it is done in all other GH workflows?

Reply:
See the code comment above ;-)

```
          # Also, we rely on the ":latest" since the datapath conformance suite
          # is heavily WIP.
```

Reply:
I think I'm fine with `latest` in the intermediate period but I would keep this job marked "optional", ie *not* blocking PRs until this is stable. If you want to mark this target "required" to merge PRs, then let's get it into a stable state and set this to a fixed tag first.

Reply:
Yep, the idea is to keep for now the DP action as optional until we it becomes stable and versioning is properly applied.